lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, toastContainerRef, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [animate, setAnimate] = useState(false),
+      classes = classnames('nx-toast', 'animate', className, toastClass,
+          {'slide-in': animate},
+          {'slide-out': !animate}
+      ),
+      toastContext = useContext(ToastContext),
+      closeBtnRef = useRef<HTMLButtonElement>(null);
+
+  //Fire effect to focus on the close button of the toast that is rendered
+  useEffect(() => {
+    const closeBtn = closeBtnRef.current;
+    if (closeBtn) {
+      closeBtn.focus();
+    }
+
+    //When component is mounted, add class "animate slide-in" to trigger animation
+    setAnimate(true);
+  }, []);
+
+  //Effect to shift focus to the next toast's close button
+  //Dependency on the global toasts array
+  useEffect(() => {
```

Comment:
It might make more sense to manage this in `NxToastContainer` or `NxToastProvider`.  That would avoid the need to pass the `toastContainerRef` down here and would also avoid this function getting one multiple times in proportion with the number of active toasts

Reply:
> It might make more sense to manage this in `NxToastContainer` or `NxToastProvider`. That would avoid the need to pass the `toastContainerRef` down here and would also avoid this function getting one multiple times in proportion with the number of active toasts

Emily and I agree. We moved the code in this commit: [747a29e](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/747a29edcade1bada310a54ec1ff12490ac01388)